rcljava/src/main/cpp/org_ros2_rcljava_action_ActionServerImpl.cpp

```diff
@@ -363,3 +364,115 @@ JNICALL Java_org_ros2_rcljava_action_ActionServerImpl_nativeCheckGoalExists(
 
   return exists;
 }
+
+JNIEXPORT void
+JNICALL Java_org_ros2_rcljava_action_ActionServerImpl_nativePublishStatus(
+  JNIEnv * env, jclass, jlong jaction_server)
+{
+  assert(0 != jaction_server);
+  auto * action_server = reinterpret_cast<rcl_action_server_t *>(jaction_server);
+  rcl_action_goal_status_array_t status_message =
+    rcl_action_get_zero_initialized_goal_status_array();
+  rcl_ret_t ret = rcl_action_get_goal_status_array(action_server, &status_message);
+  if (ret != RCL_RET_OK) {
+    std::string msg = \
+      "Failed to get goal status array: " + std::string(rcl_get_error_string().str);
+    rcl_reset_error();
+    rcljava_throw_rclexception(env, ret, msg);
+    return;
+  }
+
+  ret = rcl_action_publish_status(action_server, &status_message);
+
+  if (ret != RCL_RET_OK) {
+    std::string msg = \
+      "Failed to publish status array: " + std::string(rcl_get_error_string().str);
+    rcl_reset_error();
+    rcljava_throw_rclexception(env, ret, msg);
+    return;
+  }
+}
+
+JNIEXPORT void
+JNICALL Java_org_ros2_rcljava_action_ActionServerImpl_nativePublishFeedbackMessage(
+  JNIEnv * env, jclass, jlong jaction_server, jobject jfeedback_msg,
+  jlong jfeedback_from_java, jlong jfeedback_destroy)
+{
+  assert(0 != jaction_server);
+  assert(nullptr != jfeedback_msg);
+  assert(0 != jfeedback_from_java);
+  assert(0 != jfeedback_destroy);
+
+  auto * action_server = reinterpret_cast<rcl_action_server_t *>(jaction_server);
+  auto destroy_feedback = reinterpret_cast<destroy_ros_message_signature>(jfeedback_destroy);
+  auto from_java_feedback = reinterpret_cast<convert_from_java_signature>(jfeedback_from_java);
+  void * feedback = from_java_feedback(jfeedback_msg, nullptr);
+  // can the from_java function fail?
+  // if (!feedback) {
+  // }
+  rcl_ret_t ret = rcl_action_publish_feedback(action_server, feedback);
+  destroy_feedback(feedback);
+  if (RCL_RET_OK != ret) {
+    RCLJAVA_COMMON_THROW_FROM_RCL(env, ret, "Failed to publish feedback");
+    return;
+  }
```

Comment:
Note, the macro also checks `RCL_RET_OK != ret` and has a `return` statement. So we could reduce this to:
```suggestion
  RCLJAVA_COMMON_THROW_FROM_RCL(env, ret, "Failed to publish feedback");

```